devito/at_controller.py

```diff
@@ -0,0 +1,211 @@
+import random
+from os import mkdir, path
+
+import logger
+
+
+class AutoTuner(object):
+
+    def __init__(self, op, blocked_dims=None, at_report_dir=None):
+        """Object responsible for auto tuning block sizes.
+
+        :param op: Operator object.
+        :param blocked_dims: list indicating which dims are blocked and we want to tune
+                             Default all except inner most.
+        :param at_report_dir: string - indicating path to auto tuning report directory.
+                              If not set current directory is used
+        :param blocked_dims: list indicating which dimensions we want to auto tune
+        :raises ValueError: if operator is not of Operator type
+        """
+
+        from operator import Operator
```

Comment:
It's because you are using relative imports, which is shadowed by Python's builtin `operator` module. This is the reason why relative imports are evil.